plugins/module_utils/controller.py

```diff
@@ -387,13 +387,14 @@ def delete_if_needed(
     def copy_if_needed(
         self,
         name: str,
-        copy_from: str,
+        copy_from_name: str,
+        copy_from_id: Union[int, Any],
         endpoint: str,
         item_type: str = "unknown",
     ) -> dict[str, bool]:
         response = None
-        if not copy_from:
-            msg = f"Unable to copy {item_type}, missing copy_from parameter"
+        if not copy_from_id:
```

Comment:
Why duplicate the params? If you are going to refactor this, why not just use the id?